cmake/Codegen.cmake

```diff
@@ -53,35 +53,35 @@ install(FILES ${CMAKE_BINARY_DIR}/caffe2/core/macros.h
 
 # ---[ ATen specific
 if (NOT BUILD_ATEN_MOBILE)
-  SET(OPT_FLAG "-O3 ")
-  IF(MSVC)
-    SET(OPT_FLAG "/Ox /fp:strict ")
-  ENDIF()
-  SET(VCOMP_LIB "vcomp")
+  if(MSVC)
+    set(OPT_FLAG "/OX /fp:strict ")
+  else()
+    set(OPT_FLAG "-O3 ")
+  endif()
+  SET(VCOMP_LIB vcomp$<$<CONFIG:Debug>:d>)
 
   IF("${CMAKE_BUILD_TYPE}" MATCHES "Debug")
     SET(OPT_FLAG " ")
-    SET(VCOMP_LIB "vcompd")
   ENDIF()
 
   IF(C_AVX_FOUND)
     IF(MSVC)
-      SET_SOURCE_FILES_PROPERTIES(${CMAKE_CURRENT_LIST_DIR}/../aten/src/TH/vector/AVX.cpp PROPERTIES COMPILE_FLAGS "${OPT_FLAG}/arch:AVX ${CXX_AVX_FLAGS}")
+      set_property(SOURCE ${CMAKE_CURRENT_LIST_DIR}/../aten/src/TH/vector/AVX.cpp APPEND PROPERTY CMAKE_CXX_FLAGS ${OPT_FLAG} /arch:AVX ${CXX_AVX_FLAGS})
```

Comment:
And those changes are causing the build failures on Linux.

Reply:
Thanks. I'll get build working on Linux and then send again.